src/rs/lib.rs

```diff
@@ -874,32 +870,27 @@ extern "C" fn raw_conn_callback(
 }
 
 impl Connection {
-    pub fn new() -> Connection {
-        Connection {
-            handle: ptr::null_mut(),
-        }
-    }
-
-    pub fn open<F>(&mut self, registration: &Registration, handler: F) -> Result<(), Status>
+    pub fn open<F>(registration: &Registration, handler: F) -> Result<Self, Status>
```

Comment:
Since the convention in Rust is generally to use `new` to create an object, do you think that calling the function `new` instead of `open` would make sense?

In C, since we manipulate handles, "open" works well, but in Rust, we create and return the object, so it looks like what a `new` typically does.